src/integrationTests/options.test.ts

```diff
@@ -209,11 +219,332 @@ describe('useCookie option', () => {
+      });
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+      });
+      uid2.init({
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+      });
+    });
+    test('should return next two init calls without changing anything', () => {
+      expect(getUid2LocalStorage().advertising_token).toBe(identity.advertising_token);
+      let storageConfig = getConfigStorage();
+      expect(storageConfig).toBeInstanceOf(Object);
+      expect(storageConfig).toHaveProperty('cookiePath', cookiePath);
+    });
+  });
+
+  describe('new base URL is given', () => {
+    const oldBaseUrl = baseUrl;
+    const newBaseUrl = 'http://example';
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: oldBaseUrl,
+        useCookie: true,
+      });
+    });
+    test('should use new base url', () => {
+      uid2.init({
+        baseUrl: newBaseUrl,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', newBaseUrl);
+    });
+
+    test('should use old base url', () => {
+      uid2.init({
+        cookiePath: cookiePath,
+      });
+      const configCookie = getConfigCookie();
+      expect(configCookie).toHaveProperty('baseUrl', oldBaseUrl);
+    });
+  });
+
+  describe('new identity provided but expired', () => {
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() - 100000 });
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: identity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+    test('should set value to old identity', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(identity.advertising_token);
+    });
+    test('should set refresh timer once', () => {
+      expect(setTimeout).toHaveBeenCalledTimes(1);
+      expect(clearTimeout).not.toHaveBeenCalled();
+    });
+    test('old ideneity should be in available state', () => {
+      (expect(uid2) as any).toBeInAvailableState(identity.advertising_token);
+    });
+  });
+
+  describe('new identity provided but expires before old identity', () => {
+    const oldIdentity = makeIdentity({ refresh_expires: Date.now() + 5000 });
+    const newIdentity = makeIdentity({ refresh_expires: Date.now() + 100000 });
+    const useCookie = true;
+
+    beforeEach(() => {
+      uid2.init({
+        callback: callback,
+        identity: oldIdentity,
+        baseUrl: baseUrl,
+        cookiePath: cookiePath,
+        useCookie: useCookie,
+      });
+      uid2.init({
+        identity: newIdentity,
+      });
+    });
+
+    test('should set value', () => {
+      expect(getUid2(useCookie).advertising_token).toBe(oldIdentity.advertising_token);
```

Comment:
_makeIdentity_ uses a default value for the advertising token, so they'll both be the same value, and you can't tell which one got used.

Reply:
You can override it the same way you're overriding refresh_expires.